### DISCUSSION TITLE
Micro second and nano second precision on stored in custom timestamp field

### DISCUSSION THREAD

**Discussion by ShivaniSK** [finos/TimeBase-CE#297] [Q&A]
Hi @alex-karpovich 
We currently have data in below format where timestamp is having microseconds / nanoseconds data.

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/kanashia/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/kanashia/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">

timestamp | symbol | price
-- | -- | --
2024-10-10T08:20:23.632832Z | ABC | 1
2024-10-10T08:20:23.632913Z | ABC | 2
2024-10-10T08:20:23.633106Z | ABC | 1

</body>

</html>

On executing below query over the data

```
select LAST{} (price) as priceAgg, LAST{} (timestamp) as timestampOriginal   
            from market_trades OVER TIME(1ms) 
            WHERE timestamp BETWEEN '2024-10-10 08:20:23.632'd and '2024-10-10 08:20:23.633'd 
and symbol == 'ABC' AND price is NOT NULL
```

Expected output:
<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/kanashia/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/kanashia/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">

timestamp | price | timestampOriginal
-- | -- | --
2024-10-10T08:20:23.633Z | 2 | 2024-10-10T08:20:23.632913Z
2024-10-10T08:20:23.634Z | 1 | 2024-10-10T08:20:23.633106Z

</body>

</html>

Actual Output:

<html xmlns:v="urn:schemas-microsoft-com:vml"
xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/kanashia/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/kanashia/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#0563C1" vlink="#954F72">

timestamp | price | timestampOriginal
-- | -- | --
2024-10-10T08:20:23.633Z | 2 | 2024-10-10T08:20:23.632Z
2024-10-10T08:20:23.634Z | 1 | 2024-10-10T08:20:23.633Z

</body>

</html>

As can be seen, the "timestampOriginal" column is not giving the complete microsecond level time. Its giving data only upto milliseconds. 

Can you please help us out.
We want to have the original precision of timestamp, even when we are performing aggregations.

Regards,
Shivani

**Comment by alex-karpovich** [comment]
please share schema of the stream

**Reply by ShivaniSK** [reply]
Hi @alex-karpovich ,
The timestamps mentioned above looks like "2024-10-10T08:20:23.632832Z" 10/10/2024 08:20:23.632832 AM on the timebase webadmin.

the market_trades has below columns 
-> timestamp is the TimeBase internal field
-> symbol is Timebase internal field
-> price is FLOAT DECIMAL64

**Comment by alex-karpovich** [comment]
Please use temp workaround
```
select LAST{} (price) as priceAgg, LAST{} (timestampNs) as timestampOriginal   
            from market_trades OVER TIME(1ms) 
            WHERE timestamp BETWEEN '2024-10-10 08:20:23.632'd and '2024-10-10 08:20:23.633'd 
and symbol == 'ABC' AND price is NOT NULL
```

in next version it will be fixed
